### PR TITLE
adding SLD hyphen check

### PR DESCRIPTION
**Pull request by kumarde**

### CODE REVIEW

**Comment by zakird** [comment]
This should probably return `NA` not `PASS` if the certificate doesn't have a domain name (e.g., a CA certificate).

This problem seems to exist in a number of these lints that are checking DNSName validity. Does it make sense to create a helper that can be called in `CheckApplies` to determine whether we should be checking domain validity in a give cert?

**Comment by kumarde** [comment]
@zakird I think this can be fixed by checking `IsSubscriberCert` for all the DNS checks. I'll add these to all the outstanding PRs - granted there aren't situations where CA certs have domain names 
 somewhere that need to be checked?

**Comment by zakird** [comment]
That sounds reasonable.

File: lints/lint_dnsname_hyphen_in_sld.go [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+
+func (l *DNSNameHyphenInSLD) Initialize() error {
+	return nil
+}
+
+func (l *DNSNameHyphenInSLD) CheckApplies(c *x509.Certificate) bool {
+	return util.IsSubscriberCert(c) && util.DNSNamesExist(c)
+}
+
+func hyphenInSLD(domain string) (bool, ResultStruct) {
```
Comment (zakird) [inline lints/lint_dnsname_hyphen_in_sld.go]: I'd like this to return an Error not a Result struct.

File: lints/lint_dnsname_hyphen_in_sld.go [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+
+func (l *DNSNameHyphenInSLD) Initialize() error {
+	return nil
+}
+
+func (l *DNSNameHyphenInSLD) CheckApplies(c *x509.Certificate) bool {
+	return util.IsSubscriberCert(c) && util.DNSNamesExist(c)
+}
+
+func hyphenInSLD(domain string) (bool, error) {
```
Comment (zakird) [inline lints/lint_dnsname_hyphen_in_sld.go]: This isn't what this function actually tests.

**Pull request merged**